### PR TITLE
Buttons

### PR DESCRIPTION
**Pull request by GeorgeJegon**
- style guide buttons
- `map-deep-get` mixin 

Here a preview of Buttons Documentation generate by DocumentJS:

gaiden-css-buttons

### CODE REVIEW

File: src/scss/settings/_buttons.scss [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+  big: (
+    fontSize: 16,
+    padding: (21 42)
+  )
+);
+
+// Types
+$buttonTypes: (
+  high-contrast: (
+    normal: (color: punch),
```
Comment (eduardojmatos) [inline src/scss/settings/_buttons.scss]: @GeorgeJegon can we change `normal` for `default`? I think that is more understandable than `normal`.

File: src/scss/tools/mixins/_get_color.scss [hunk 2]
```diff
@@ -0,0 +1,7 @@
+@function get-color($color, $key: base, $opacity: 1, $palette: $colors) {
+  @if type-of($color) == 'map' {
+    @return get-color($color...);
+  }
+  $colorFromPalette: map-get(map-get($palette, $color), $key);
```
Comment (eduardojmatos) [inline src/scss/tools/mixins/_get_color.scss]: @GeorgeJegon let's start to use `map-deep-get` \o/ 😺

File: src/scss/tools/mixins/_map_deep_get.scss [hunk 3]
```diff
@@ -0,0 +1,12 @@
+/// Map deep get
+/// @author Hugo Giraudel
+/// @access public
+/// @param {Map} $map - Map
+/// @param {Arglist} $keys - Key chain
+/// @return {*} - Desired value
```
Comment (eduardojmatos) [inline src/scss/tools/mixins/_map_deep_get.scss]: @GeorgeJegon this comment and description of mixin is in the format of documentjs?

Comment (GeorgeJegon) [inline src/scss/tools/mixins/_map_deep_get.scss]: @eduardojmatos what do you think about we adjust this documentation after we create others components

Comment (eduardojmatos) [inline src/scss/tools/mixins/_map_deep_get.scss]: @GeorgeJegon no problem, I totally agree.

**Pull request merged**